pkg/plugin/vector/store/store.go

```diff
@@ -0,0 +1,70 @@
+package store
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+)
+
+type VectorStoreType string
+
+const (
+	VectorStoreTypeQdrant           VectorStoreType = "qdrant"
+	VectorStoreTypeGrafanaVectorAPI VectorStoreType = "grafana/vectorapi"
+)
+
+type Collection struct {
+	Name      string `json:"name"`
+	Dimension int    `json:"dimension"`
+	Model     string `json:"model"`
```

Comment:
Not sure about this particular bit, the model name isn't stored in most vector dbs. For simplicity, 
- we could go with a single model (defined at the embedder level, not at the store level)
- we could also have a separate method to get the model for a given collection
  - this could work with the single defined model at the embedder level, and then overrides based on a file for example
  - alternatively, this could call out an API for supported vector stores, that can return the model that was used to generate the embeddings (again, very few support at the collection level)

Reply:
After chatting we decided to nest the settings a bit more and store a single model at the `vector` level, with `store` and `embed` settings nested beneath that. So it looks something like:

```yaml
jsonData:
  vector:
    model: text-embedding-ada-002
    embed:
      type: openai
    store:
      type: qdrant
      qdrant:
        address: qdrant:6334
```